src/utils.js

```diff
@@ -99,31 +99,48 @@ const isLicenseError = (license = '') => licensesExceptions.includes(license);
  * If a package license does not satisfy the allowed SPDX id list, the package will be included on the "forbidden" list.
  * @param {string} expression - A SPDX expression
  * @param {object[]} packages - A list of packages to be checked against the SPDX expression
+ * @param {boolean} allow - Determines the license check. If true, forbidden will contain all the packages that
+ * do not comply with the expression. If false, forbidden will contain all the packages that comply with the expression
  * @return {{forbidden: object[], nonCompliant: object[]}} - A couple of lists including the packages that satisfy the SPDX expression
  * and the packages with a non SPDX compliant license
  */
-const checkPackagesLicenses = (expression, packages) => {
-  const validSpdxIds = expression && spdxIds.filter(id => !isLicenseError(id) && !satisfiesSPDXLicense(id, expression)); // @TODO Refactor after issue has been solved
-  const allowedLicensesExp = expression && generateSPDXExpression(validSpdxIds);
+const checkPackagesLicenses = (expression, packages, allow = false) => {
+  const validSpdxIds = spdxIds.filter(id => {
```

Comment:
@inigomarquinez here we filter out the licenses depending on the `allow` flag. If it is true, it will yield only the licenses that matches the input expression. If it false, it will yield any other licenses that don't match the input expression. For instance:

- `allowOnly` has value `MIT`. The result will only include `MIT`
- `failOn` has value `MIT`. The result will include all the licenses **except** `MIT`

That's why on line 121 we alway check whether the package license satisfies the result of the previous list.

Reply:
Understood. Thanks!